### PR TITLE
Add: more textual formats support

### PR DESCRIPTION
**Pull request by Fraggle**
## Description

Add support for:

```json
{
  "text/html": [".html", ".htm", ".xhtml", ".xhtml+xml"],
  "text/xml": [".xml"],
  "text/calendar": [".ics"],
  "text/css": [".css"],
  "text/javascript": [".js", ".mjs"],
  "application/json": [".json"],
  "application/xml": [".xml"],
  "application/x-sh": [".sh"],
}
```

(more to come in subsequent PR)

## Risk

Low (tested locally)

## Deploy Plan

Deploy `front`

### CODE REVIEW

File: front/lib/api/files/upsert.ts [hunk 1]
```diff
@@ -66,37 +67,45 @@ async function generateSnippet(
...
+    );
+  }
 
-      return new Ok(snippet);
-    case "text/markdown":
-    case "text/plain":
-    case "text/vnd.dust.attachment.slack.thread":
+  if (isSupportedDelimitedTextContentType(file.contentType)) {
+    const format =
+      file.contentType === "text/csv" ||
```
Comment (spolu) [inline front/lib/api/files/upsert.ts:78]: this is a bit brittle? should we fetch the extension based on the content type since we capture that information

Comment (spolu) [reply front/lib/api/files/upsert.ts:78]: We could also just use the content type here

**Review by spolu — APPROVED** [review]
LGTM

Maybe we should have equivalent of `isSupportedDelimitedTextContentType` for all groups of content types so that adding a new one is just one line without having to check all the switch cases?

We could do in most places:

```
if (isSupportedImageContentType) { ... return }
if (isSupportedDelimitedTextContentType) { ... return }
if (isSupportedTextContentType) { ... return }
if (isSupportedCodeContentType) { ... return }
if (isSupportedDataContentType)  { ... return }

switch (...) {
  default:
    assertNever
}
```

`isSupportedCodeContentType` for css js etc... and we can add much more of course
`isSupportedDataContentType` for json/xml I presume maybe ics?

**Comment by Fraggle** [comment]
>Maybe we should have equivalent of isSupportedDelimitedTextContentType for all groups of content types so that adding a new one is just one line without having to check all the switch cases?

Good idea, will do in another PR :)

**Pull request merged**